tikv-client/src/main/java/com/pingcap/tikv/TiSession.java

```diff
@@ -141,6 +143,25 @@ public PDClient getPDClient() {
     return res;
   }
 
+  public TiTimestamp getSnapshotTimestamp() {
+    return snapshotTimestamp == null ? getTimestamp() : snapshotTimestamp;
+  }
+
+  public Snapshot createSnapshotWithSnapshotTimestamp() {
```

Comment:
private？

Reply:
If you want to use a singleton, make sure this method is private in case someone misuses it.

Reply:
SnapshotCatalog is a singleton, the snapshot is not a singleton, It will return a new snapshot in every call.
I refer to the method`createSnapshot()` which is also public